tests/test_flow/test_persistence_fuzz.py

```diff
@@ -270,17 +311,30 @@ def run(self, n_iterations):
                     self.model.expected_entity_value(affected_name)
                 )
 
-            if make_func_change or self._versioning_mode == 'auto':
-                expected_called_names = self.model.entity_names(
+            if make_func_change:
+                entity_names_in_between = self.model.entity_names(
                     downstream_of=updated_name,
                     upstream_of=affected_names,
                 )
-                assert (
-                    list(sorted(self.model.called_entity_names())) ==
-                    list(sorted(expected_called_names))
-                )
+                entity_names_in_between = set(entity_names_in_between)
+                nondeterministic_ancestors = set()
+                for entity_name in entity_names_in_between:
+                    nondeterministic_ancestors |=\
+                        self.model.nondeterministic_upstream_entity_names(entity_name)
+
+                expected_called_names = entity_names_in_between | nondeterministic_ancestors
+                actual_called_names = self.model.called_entity_names()
+
+                # TODO also check that the results with multiple calls are all nondeterminstic
+                assert set(actual_called_names) == expected_called_names
             else:
-                assert len(self.model.called_entity_names()) == 0
+                actual_called_names = self.model.called_entity_names()
+                expected_called_names =\
+                    self.model.nondeterministic_upstream_entity_names(affected_names)
+                if self._versioning_mode == 'auto':
+                    expected_called_names.add(updated_name)
```

Comment:
Shouldn't we be adding every entity in between the updated and affected nodes?

Reply:
Not when there is a non functional change since we now use value hash instead of provenance in caching. 

This is honestly on me. When I made changes to this test with the value hash change, I didn't test for this case. I realized that when I really had to go through the test for this change.

Reply:
Also, given that even you missed it, this probably warrants a comment. I'll add one.

Reply:
Oh right! That makes sense.

So, does this mean the test code on master is currently wrong? Is it surprising that we didn't hit this case?